extensions/common/sql/sql-core/src/main/java/org/eclipse/edc/sql/translation/JsonFieldTranslator.java

```diff
@@ -71,4 +63,31 @@ public WhereClause toWhereClause(List<PathItem> path, Criterion criterion, SqlOp
         );
     }
 
+    private String checkStatementByType(Class<?> type, String statement) {
```

Comment:
nit: method name is not correct. It doesn't "check" anything, it creates a typecast expression based on a type. I would suggest a name like `createStatementForType` or similar.

Reply:
good point, thank you, changed in https://github.com/eclipse-edc/Connector/pull/4435/commits/8196e469c2dd48874989ac12188bbcb6757909cc